minio/minio
# Add support for multiple OpenID providers with role policies

## Description

- When using multiple providers, claim based providers are not allowed. All
providers must use role policies.

Console changes are not done in this change - so it is not yet possible to login via the web console.

## Motivation and Context

Extend role policy support to enable multiple OpenID provider/apps.

## How to test this PR?

See examples in the doc update in the PR. 

Test setup can be done with dex using the containers and make file in https://github.com/minio/minio-iam-testing - see readme: https://github.com/minio/minio-iam-testing/tree/main/openid#running-minio-with-multiple-oidc-providers

STS credentials can be generated programmatically with the program at `/docs/site-replication/gen-oidc-sts-cred.go` from this PR.

## Types of changes

## Checklist:
